tools/config_generator/config_generator.cpp

```diff
@@ -0,0 +1,811 @@
+                                             std::get<double>(parsed_val));
+                } else if(std::holds_alternative<std::string>(parsed_val)) {
+                    set_param_to_config_file(
+                        config_map_it->first,
+                        std::get<std::string>(parsed_val));
+                } else {
+                    __builtin_unreachable();
+                }
+            }
+            config_map_it++;
+        }
+
+        auto randomize_int = find_value<std::size_t>(tmpl_randomize_values);
+        m_randomize = randomize_int == 1;
+        if(m_randomize) {
+            srand(std::chrono::system_clock::now().time_since_epoch().count());
+            generator.seed(
+                std::chrono::system_clock::now().time_since_epoch().count());
+        } else {
+            generator.seed(1);
+            srand(1);
+        }
+
+        // Create Components
+        const auto is_two_phase_mode
+            = get_param_from_template_file(two_phase_mode, config_map);
+        const auto shard_count
+            = get_param_from_template_file(shard_count_key, config_map);
+        const auto sentinel_count
+            = get_param_from_template_file(sentinel_count_key, config_map);
+
+        auto shard_size = find_value<std::size_t>(tmpl_shard_size);
+        // Add one since this is 255 and we are 0 indexing but want
+        // 0-255 inclusive
+        calculate_shard_coverage(std::get<size_t>(shard_count),
+                                 shard_size.value() + 1);
+
+        if(std::holds_alternative<size_t>(is_two_phase_mode)
+           && std::get<size_t>(is_two_phase_mode) == 1) {
+            const auto coordinator_count
+                = get_param_from_template_file(coordinator_count_key,
+                                               config_map);
+            return_msg += create_component(sentinel_count_key,
+                                           std::get<size_t>(sentinel_count),
+                                           true);
+            return_msg += create_component(shard_count_key,
+                                           std::get<size_t>(shard_count),
+                                           true);
+            return_msg += create_component(coordinator_count_key,
+                                           std::get<size_t>(coordinator_count),
+                                           true);
+        } else {
+            const auto atomizer_count
+                = get_param_from_template_file(atomizer_count_key, config_map);
+            const auto archiver_count
+                = get_param_from_template_file(archiver_count_key, config_map);
+            const auto watchtower_count
+                = get_param_from_template_file(watchtower_count_key,
+                                               config_map);
+            return_msg += create_component(shard_count_key,
+                                           std::get<size_t>(shard_count),
```

Comment:
If `shard_count` is not specified in the template file, or is initialized to anything besides a legal `size_t` value, `std::get<size_t>(shard_count)` will fail and cause an abort. This also applies to all below calls of `create_component` with their corresponding component names.

Since this is a tool external to the system, this is probably not a big deal, but I am putting this here for consideration.

Reply:
Yeah, I'm also not too concerned about this. Though, anything that's a bit error-prone like this underscores that everything here would be less error-prone if we had better error-handling for config file parsing.